### DISCUSSION TITLE
Wrap long strings to fit in a box

### DISCUSSION THREAD

**Discussion by dracir9** [Bodmer/TFT_eSPI#2819] [Q&A - Fonts]
Hi,

Is there any option to wrap long string inside a box? I would like to define an area and then draw a string inside. If the string doesn't fit horizontally it should automatically split the string and add the remaining in new line.
That would be useful for building buttons or text boxes with dynamic text where the number of characters is not known.

I've tryed using the viewport function, but the text is simply clipped when it goes off screen

**Comment by Bodmer** [comment]
There is no specific function for that. The viewport acts a a clipping window and wraping will ONLY occur if you use the print class. e.g.
```
  tft.setTextWrap(true, false); // Wrap in x but not y
  tft.setCursor(0,0);
  tft.print( "This is a very long string");
```
You may need to write a dedicated function to split the line at a "space" and render the line parts at the datum of your choosing.
